packages/block-library/src/tabs/view.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * WordPress dependencies
+ */
+import { store, getContext, getElement } from '@wordpress/interactivity';
+
+/**
+ * Determines whether a value is numeric.
+ *
+ * @param {*} value A vlue to check.
+ * @return {boolean} Whether the value is numeric.
+ */
+function isNumeric( value ) {
+	return ! isNaN( parseFloat( value ) );
+}
+
+/**
+ * Generates a map of tab panel IDs to tab label IDs.
+ *
+ * @param {Element[]} tabLabels List of tab label elements.
+ * @return {Map} A map of tab panel IDs to tab label IDs.
+ */
+function getTabPanelToLabelIdMap( tabLabels ) {
+	if ( ! tabLabels ) {
+		return new Map();
+	}
+
+	return new Map(
+		tabLabels.map( ( tabLabel ) => {
+			const tabPanelId = tabLabel.getAttribute( 'href' )?.substring( 1 );
+			const tabLabelId = tabLabel.getAttribute( 'id' );
+			return [ tabPanelId, tabLabelId ];
+		} )
+	);
+}
+
+/**
+ * Initializes the tabs block with the necessary ARIA attributes.
+ *
+ * @param {HTMLElement} ref The block wrapper element.
+ */
+function initTabs( ref ) {
+	// Add class interactive to the block wrapper to indicate JavaScript has been loaded.
+	ref.classList.add( 'interactive' );
+
+	const tabList = ref.querySelector( '.wp-block-tabs__list' );
+	if ( tabList ) {
+		// Add role="tablist" to the <ul> element.
+		tabList.setAttribute( 'role', 'tablist' );
+
+		// Use the hidden <h3>Contents</h3> element as the label for the tab list, if found.
+		const tabListLabel = ref.querySelector( '.wp-block-tabs__title' );
+		if ( tabListLabel ) {
+			const tabListLabelId = tabListLabel.id || 'tablist-label';
+			tabListLabel.id = tabListLabelId;
+			tabList.setAttribute( 'aria-labelledby', tabListLabelId );
+		} else {
+			tabList.setAttribute( 'aria-label', 'Tabs' );
```

Comment:
Any strings should be translatable. Also again the question about not using directives here :)

Reply:
This has been removed in https://github.com/WordPress/gutenberg/pull/63689/commits/cc304b55ad47d7b63985e39d965f420023c2f44d